ledger/accountdb.go

```diff
@@ -1989,6 +1995,43 @@ func (qs *accountsDbQueries) lookupKeyValue(key string) (pv persistedValue, err
 	return
 }
 
+func (qs *accountsDbQueries) lookupKeysByPrefix(
+	prefix string, maxKeyNum uint64, sawDeletedKeys map[string]bool, keysWithValue *map[string]bool,
```

Comment:
I pass in by pointer, then the implementation in `accountdb` can directly add keys to the set, rather than returning a set of keys and append later.

Reply:
Yes, I know you were using it as a set, and ignoring the boolean. But my idea was to _use_ the boolean. Storing a `false` would mean you have seen a delete, storing a `true` would mean that you have seen a value. You never add a key that already has an entry, regardless of whether the entry is true or false. (Because you want the first one you find while walking backwards.  Then you report only the entries that have a `true` back to the REST API.

So now you just pass a single `map[string]bool` rather than two, perhaps called results.  So you pass in a `results` map containing the seen deleted keys (it can also contain the keys you've seen that _have_ been set, so you don't bother accumulating them).

Since maps are reference types, you can add to them in a function without passing them by pointer. (Think about how you can pass in a slice by value, but still alter a value in it.)

Reply:
got it got it, it implies that I don't have to maintain 2 sets, but only 1, by piggybacking the map-to boolean value to indicate that the key's validity.

It also surprises me that `map` is reference type in golang and we just pass by instance and modify inside function.